core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+/// syntax - spacing around special sumbols and adding the missing sumbols
+/// ("std::", ",..."). "FormatMethodForDoxygen" works for the functions defined
+/// inside namespaces as well. We pass pointer to the class as well, to make
+/// sure the class exists.
+///
+/// \param[in] className <the name of the class
+/// \param[in] classVariable <pointer to the class
+/// \param[in] methodName <the name of the method from the class
 
-   Printf("\nROOT special commands.");
-   Printf("===========================================================================");
-   Printf("             pwd          : show current directory, pad and style");
-   Printf("             ls           : list contents of current directory");
-   Printf("             which [file] : shows path of macro file");
+static TString FormatMethodForDoxygen(const TString &className, TClass *classVariable, const TString &methodName)
+{
+   // With "GetSignature" we get the arguments of "methodName" and put them in a TString.
+   TString methodArguments(((TFunction *)(classVariable->GetListOfMethods()->FindObject(methodName)))->GetSignature());
+   // The TString "MethodArguments" is modified with respect of Doxygen requirements.
+   methodArguments.ReplaceAll(" = ", "=");
+   // maybe add also " =" and "= ";
+   methodArguments.ReplaceAll("* ", " *");
+   // is this always? below
+   methodArguments.ReplaceAll("*=", " *=");
+   methodArguments.ReplaceAll("*)", " *)");
+   methodArguments.ReplaceAll("*,", " *,");
+   methodArguments.ReplaceAll("*& ", " *&");
+   methodArguments.ReplaceAll("& ", " &");
+   methodArguments.ReplaceAll("ostream", "std::ostream");
+   methodArguments.ReplaceAll("istream", "std::istream");
+   methodArguments.ReplaceAll("*msgfmt", "*msgfmt,...");
+   // We need to take the regular expresion of the arguments (mostly if enumeration appears).
+   TString classNameRE ("\\b");
+   classNameRE.Append(className);
+   classNameRE.Append("::\\b");
+   TPRegexp argFix(classNameRE);
+   argFix.Substitute(methodArguments, "");
+   return methodArguments;
+}
 }
 
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+/// The function checks if a member function (methodName) of a class (className)
+/// is defined as inline. If so, it also checks if it is virtual. Then the
+/// return type of "methodName" is changed for the need of Doxygen and with respect
+/// to the function definition. The corrected "returnType" is returned.
+/// We pass also pointer to the class (classVariable) it order to asure that
+/// the class exists.
+///
+/// \param[in] className <the name of the class
+/// \param[in] classVariable <pointer to the class
+/// \param[in] methodName <the name of the method from the class
+
+static TString FormatReturnTypeForDoxygen(const TString &className, TClass *classVariable,  const TString &methodName)
+{
+   TMethod *method = (TMethod *)classVariable->GetListOfMethods()->FindObject(methodName);
```

Comment:
Consider passing this `method` as argument. If you cannot then you must check here whether the lookup succeeded - i.e. `if (!method)` then error and return.